### PR TITLE
update bugsnag-android dep to v6.8.0

### PR DESCRIPTION
**Pull request by richardelms**
## Goal

Update the bugsnag-android dependency to v6.8.0

## Changeset

- Update gradle build references to bugsnag-android to v6.8.0.
- Convert discardClasses and redactedKeys to Regex patterns when sending to the android layer.
- Remove references to ThreadType, replace required ones with refs to ErrorType.
- Convert the getThreadId and setThreadId JNI refs to the new string signatures.
- The Unreal layer no longer handles auto session tracking due to improvements in the android notifier.

## Testing

- Updated existing tests to match the changed data types.
- Updated the fixture to clear the new bugsnag cache location

### CODE REVIEW

File: Plugins/Bugsnag/Source/Bugsnag/Private/Android/JNIUtilities.h [hunk 1]
```diff
@@ -362,6 +361,16 @@ class FAndroidPlatformJNI
 	 * @return A Java object reference or null on failure
 	 */
 	static jobject ParseStringSet(JNIEnv* Env, const JNIReferenceCache* Cache, const TArray<FString>& Values);
+	/**
```
Comment (lemnik) [inline Plugins/Bugsnag/Source/Bugsnag/Private/Android/JNIUtilities.h]: nit: formatting
```suggestion
	static jobject ParseStringSet(JNIEnv* Env, const JNIReferenceCache* Cache, const TArray<FString>& Values);
	
	/**
```

File: Plugins/Bugsnag/Source/Bugsnag/Private/Android/AndroidThread.h [hunk 2]
```diff
@@ -22,17 +22,30 @@ class FAndroidThread : public IBugsnagThread, FJavaObjectWrapper
...
+		// Retrieve the thread ID as a Java string
+		jstring jId = static_cast<jstring>((*Env).CallObjectMethod(JavaObject, Cache->ThreadGetId));
+		if (FAndroidPlatformJNI::CheckAndClearException(Env) || !jId)
+		{
+			return TEXT("");
+		}
+
+		// Convert the Java string to an FString
+		const char* nativeString = (*Env).GetStringUTFChars(jId, nullptr);
+		FString Id = FString(UTF8_TO_TCHAR(nativeString));
```
Comment (lemnik) [inline Plugins/Bugsnag/Source/Bugsnag/Private/Android/AndroidThread.h]: Should we maybe be null-checking `nativeString` as well? There are conditions where `GetStringUTFChars` will return null for a valid string.

**Review by twometresteve — COMMENTED** [review]
Looks fine from my perspective.

**Review by lemnik — APPROVED** [review]
LGTM

**Pull request merged**